Source/WebCore/page/scrolling/mac/ScrollerPairMac.mm

```diff
@@ -351,21 +351,36 @@ - (void)scrollerImpPair:(NSScrollerImpPair *)scrollerImpPair updateScrollerStyle
 void ScrollerPairMac::mouseEnteredContentArea()
 {
     LOG_WITH_STREAM(OverlayScrollbars, stream << "ScrollerPairMac for [" << m_scrollingNode.scrollingNodeID() << "] mouseEnteredContentArea");
-    if ([m_scrollerImpPair overlayScrollerStateIsLocked])
-        return;
 
-    [m_scrollerImpPair mouseEnteredContentArea];
+    ensureOnMainThreadWithProtectedThis([this] {
+        if ([m_scrollerImpPair overlayScrollerStateIsLocked])
+            return;
+
+        [m_scrollerImpPair mouseEnteredContentArea];
+    });
 }
 
 void ScrollerPairMac::mouseExitedContentArea()
 {
     LOG_WITH_STREAM(OverlayScrollbars, stream << "ScrollerPairMac for [" << m_scrollingNode.scrollingNodeID() << "] mouseExitedContentArea");
-    if ([m_scrollerImpPair overlayScrollerStateIsLocked])
-        return;
+    
+    ensureOnMainThreadWithProtectedThis([this] {
+        if ([m_scrollerImpPair overlayScrollerStateIsLocked])
+            return;
 
-    [m_scrollerImpPair mouseExitedContentArea];
+        [m_scrollerImpPair mouseExitedContentArea];
+    });
 }
 
+void ScrollerPairMac::mouseMovedInContentArea()
```

Comment:
Why not use the “mouse move” terminology in the state tree?

Reply:
Fixed.